test/integration/query-interface.test.js

```diff
@@ -636,6 +636,56 @@ describe(Support.getTestDialectTeaser('QueryInterface'), () => {
         });
         let constraints = await this.queryInterface.showConstraint('posts');
         constraints = constraints.map(constraint => constraint.constraintName);
+        expect(constraints).to.include('posts_username_users_fk');
+
+        if (dialect === 'postgres') {
+          const foreignKeys = await this.sequelize.query(
+            this.queryInterface.queryGenerator.getForeignKeysQuery('posts', this.sequelize.config.database),
+            { type: this.sequelize.QueryTypes.FOREIGNKEYS }
+          );
+          expect(Object.keys(foreignKeys[0])).to.have.length(8);
+          expect(foreignKeys[0].condef).to.equal('FOREIGN KEY (username) REFERENCES users(username) ON UPDATE CASCADE ON DELETE CASCADE');
+        }
+
+        await this.queryInterface.removeConstraint('posts', 'posts_username_users_fk');
+        constraints = await this.queryInterface.showConstraint('posts');
+        constraints = constraints.map(constraint => constraint.constraintName);
+        expect(constraints).to.not.include('posts_username_users_fk');
+      });
+
+      it('should add, read & remove foreign key constraint with notValid', async function() {
+        await this.queryInterface.removeColumn('users', 'id');
+        await this.queryInterface.changeColumn('users', 'username', {
+          type: DataTypes.STRING,
+          allowNull: false
+        });
+        await this.queryInterface.addConstraint('users', {
+          type: 'PRIMARY KEY',
+          fields: ['username']
+        });
+        await this.queryInterface.addConstraint('posts', {
+          fields: ['username'],
+          references: {
+            table: 'users',
+            field: 'username',
+            notValid: true
+          },
+          onDelete: 'cascade',
+          onUpdate: 'cascade',
+          type: 'foreign key'
+        });
+        let constraints = await this.queryInterface.showConstraint('posts');
+        constraints = constraints.map(constraint => constraint.constraintName);
+
+        if (dialect === 'postgres') {
+          const foreignKeys = await this.sequelize.query(
+            this.queryInterface.queryGenerator.getForeignKeysQuery('posts', this.sequelize.config.database),
+            { type: this.sequelize.QueryTypes.FOREIGNKEYS }
+          );
```

Comment:
I am using the `condef` (`pg_catalog.pg_get_constraintdef(r.oid, true)`) to ensure that the statement that was run has the `NOT VALID` in it. I'd be curious if there another function I could use, happy to update accordingly.  Let me know.

I noticed `getForeignKeysQuery` is not supported under `getForeignKeyReferencesQuery` so didn't want to update an unrelated function in this change. 

https://github.com/sequelize/sequelize/blob/996e273ceee8ab634d99cf9576e82b76188b6725/lib/dialects/postgres/query-generator.js#L879